hmy/downloader/shortrange.go

```diff
@@ -51,21 +49,25 @@ func (d *Downloader) doShortRangeSync() (int, error) {
 	d.status.setTargetBN(expEndBN)
 	defer d.finishSyncing()
 
-	blocks, err := sh.getBlocksByHashes(hashChain, whitelist)
+	blocks, stids, err := sh.getBlocksByHashes(hashChain, whitelist)
 	if err != nil {
 		if !errors.Is(err, context.Canceled) {
 			sh.removeStreams(whitelist) // Remote nodes cannot provide blocks with target hashes
 		}
 		return 0, errors.Wrap(err, "getBlocksByHashes")
 	}
 
-	n, err := d.ih.verifyAndInsertBlocks(blocks)
+	n, err := verifyAndInsertBlocks(d.bc, blocks)
 	numBlocksInsertedShortRangeHistogramVec.With(d.promLabels()).Observe(float64(n))
 	if err != nil {
-		if !errors.As(err, &emptySigVerifyError) {
+		if sh.blameAllStreams(blocks, n, err) {
 			sh.removeStreams(whitelist) // Data provided by remote nodes is corrupted
+		} else {
```

Comment:
what's the cases when a peer is blamed? what happen after the blaming?

Reply:
?

Reply:
1. This is the case where the peer provides the last block in a batch gives a wrong commit signature. Since the commit signature is not include in block hash, it could be the case that other whitelist streams provides the correct hash chain, and only this node is giving wrong data.
2. In other cases, since all block data is deterministic with a fixed block hash, if signature verification or block insertion fails, we can imply that the given hash chain is invalid. Thus all streams providing that hash chain is guilty.
3. Currently, for the guilty screams, we simply close the connection to that stream. Will implement a timed blacklist later.